Neos.Neos/Classes/Http/ContentSubgraphUriProcessor.php

```diff
@@ -0,0 +1,117 @@
+    {
+        $uriConstraints = UriConstraints::create();
+
+        /** @var ContentContext $contentContext */
+        $contentContext = $node->getContext();
+
+        if (!$contentContext->isInBackend()) {
+            $presets = $this->dimensionPresetSource->getAllPresets();
+            $dimensionValues = $node->getContext()->getDimensions();
+            $this->sortDimensionValuesByOffset($dimensionValues, $presets);
+            $uriPathSegmentOffset = 0;
+            $uriPathSegmentConstraints = UriConstraints::create();
+            $allUriPathSegmentDetectableDimensionPresetsAreDefault = true;
+
+            foreach ($presets as $dimensionName => $presetConfiguration) {
+                $options = $presetConfiguration['resolution']['options'] ?? [];
+                $resolutionMode = new ContentDimensionResolutionMode(
+                    $presetConfiguration['resolution']['mode']
+                    ?? ContentDimensionResolutionMode::RESOLUTION_MODE_URIPATHSEGMENT
+                );
+
+                if (isset($dimensionValues[$dimensionName])) {
+                    $preset = $this->dimensionPresetSource->findPresetByDimensionValues($dimensionName, $dimensionValues[$dimensionName]);
+                } else {
+                    $preset = $presetConfiguration['presets'][$presetConfiguration['defaultPreset']];
+                    $preset['identifier'] = $presetConfiguration['defaultPreset'];
+                }
+
+                $linkProcessor = $this->dimensionPresetLinkProcessorResolver->resolveDimensionPresetLinkProcessor($dimensionName, $presetConfiguration);
+                if ($resolutionMode->getMode() === ContentDimensionResolutionMode::RESOLUTION_MODE_URIPATHSEGMENT) {
+                    if (!isset($options['offset'])) {
+                        $options['offset'] = $uriPathSegmentOffset;
+                    }
+
+                    if ($presetConfiguration['defaultPreset'] !== $preset['identifier']) {
+                        $allUriPathSegmentDetectableDimensionPresetsAreDefault = false;
+                    }
+                    $uriPathSegmentOffset++;
+                    $uriPathSegmentConstraints = $linkProcessor->processUriConstraints($uriPathSegmentConstraints, $dimensionName, $presetConfiguration, $preset, $options);
+                } else {
+                    $uriConstraints = $linkProcessor->processUriConstraints($uriConstraints, $dimensionName, $presetConfiguration, $preset, $options);
+                }
+            }
+
+            if ((!$this->supportEmptySegmentForDimensions || !$allUriPathSegmentDetectableDimensionPresetsAreDefault)
+                && $uriPathSegmentOffset > 0) {
+                $uriConstraints = $uriConstraints->merge($uriPathSegmentConstraints);
+                if ($node->getParentPath() !== SiteService::SITES_ROOT_PATH) {
+                    $uriConstraints = $uriConstraints->withPathPrefix('/', true);
```

Comment:
This line causes links in not-default dimension having double slashes in the uri after the dimension segment like `http://127.0.0.1:8082/de//features.html`. Just discovered the effect during testing and habe no opinion yet wether or not this us needed.

To reproduce this translate a sub-node in neos-demo in any language.